lib/node_modules/@stdlib/array/base/map/docs/types/index.d.ts

```diff
@@ -0,0 +1,214 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Collection, AccessorArrayLike } from '@stdlib/types/array';
+
+/**
+* Callback invoked for each array element.
+*
+* @returns result
+*/
+type Nullary<U, V> = ( this: V ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - current array element
+* @returns result
+*/
+type Unary<T, U, V> = ( this: V, value: T ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - current array element
+* @param index - current array element index
+* @returns result
+*/
+type Binary<T, U, V> = ( this: V, value: T, index: number ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns result
+*/
+type Ternary<T, U, V> = ( this: V, value: T, index: number, arr: Collection<T> | AccessorArrayLike<T> ) => U;
+
+/**
+* Callback invoked for each array element.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param arr - input array
+* @returns result
+*/
+type Callback<T, U, V> = Nullary<U, V> | Unary<T, U, V> | Binary<T, U, V> | Ternary<T, U, V>;
+
+/**
+* Interface describing the main export.
+*/
+interface Routine {
+	/**
+	* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+	*
+	* @param x - input array
+	* @param fcn - callback function
+	* @param thisArg - callback execution context
+	* @returns output array
+	*
+	* @example
+	* var ones = require( '@stdlib/array/base/ones' );
+	*
+	* function scale( v ) {
+	*     return v * 10;
+	* }
+	*
+	* var x = ones( 4 );
+	* var y = map( x, scale );
+	* // returns [ 10, 10, 10, 10 ]
+	*/
+	<T = unknown, U = unknown, V = unknown>( x: Collection<T>, fcn: Callback<T, U, V>, thisArg?: ThisParameterType<Callback<T, U, V>> ): Collection<U>;
+
+	/**
+	* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+	*
+	* @param x - input array object
+	* @param fcn - callback function
+	* @param thisArg - callback execution context
+	* @returns output array object
+	*
+	* @example
+	* var toAccessorArray = require( '@stdlib/array/to-accessor-array' );
+	* var ones = require( '@stdlib/array/base/ones' );
+	*
+	* function scale( v ) {
+	*     return v * 10;
+	* }
+	*
+	* var x = toAccessorArray( ones( 4 ) );
+	* var y = map( x, scale );
+	* // returns {...}
+	*/
+	<T = unknown, U = unknown, V = unknown>( x: AccessorArrayLike<T>, fcn: Callback<T, U, V>, thisArg?: ThisParameterType<Callback<T, U, V>> ): AccessorArrayLike<U>;
```

Comment:
@headlessNode You need to switch the order of the overloads, as an `AccessorArrayLike` object is also a collection.